app/src/main/java/org/mozilla/tv/firefox/webrender/cursor/CursorViewModel.kt

```diff
@@ -63,63 +49,8 @@ class CursorViewModel(
-        validateMotionEvent(action, SystemClock.uptimeMillis(), pos)?.use {
-            _touchSimulationLiveData.value = Consumable.from(it)
-        }
-    }
-
-    /**
-     * Problem: there is a distinctive time difference between [MotionEvent.ACTION_DOWN] and
-     * [MotionEvent.ACTION_UP] when using DPAD_SELECT_KEY. [GeckoSession] requires them to have matching
-     * [MotionEvent.getEventTime] in order to register such [KeyEvent] as "Click" touch event.
-     *
-     * [validateMotionEvent] checks for previous ACTION_DOWN vs. upcoming ACTION_UP to calculate
-     * the diff against [ViewConfiguration.getLongPressTimeout] to determine whether "Click" or
-     * "Long Press" event. Also filters unnecessary [MotionEvent] that occurs when holding down
-     * DPAD_SELECT_KEY (see (*)) by bypassing consecutive ACTION_DOWN calls.
-     *
-     * (*) Capture the first instance of [KeyEvent.ACTION_DOWN] upon long press and check against
-     * the first instance of [KeyEvent.ACTION_UP].
-     */
-    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    @Synchronized
-    fun validateMotionEvent(action: Int, eventTime: Long, pos: PointF): MotionEvent? {
-        val downTime = eventTime - DOWN_TIME_OFFSET_MILLIS
-
-        when (action) {
-            KeyEvent.ACTION_DOWN -> {
-                prevDownMotionEvent?.let {
-                    if (eventTime - it.eventTime >= DPAD_LONG_PRESS_TIMEOUT) {
-                        prevDownMotionEvent = MotionEvent.obtain(downTime, eventTime, action, pos.x, pos.y, 0)
-                    } else {
-                        return null
-                    }
-                } ?: let {
-                    prevDownMotionEvent = MotionEvent.obtain(downTime, eventTime, action, pos.x, pos.y, 0)
-                }
-
-                return prevDownMotionEvent
-            }
-            KeyEvent.ACTION_UP -> {
-                prevDownMotionEvent?.let {
-                    if (eventTime - it.eventTime < DPAD_TAP_TIMEOUT) {
-                        // Register it as a "Click" Event and match the down & event time with the previous event
-                        return MotionEvent.obtain(it.downTime, it.eventTime, action, pos.x, pos.y, 0)
-                    } else if (eventTime - it.eventTime < DPAD_LONG_PRESS_TIMEOUT) {
-                        return MotionEvent.obtain(downTime, eventTime, action, pos.x, pos.y, 0)
-                    }
-
-                    prevDownMotionEvent = null
-                } ?: return null
-            }
-            else -> {
-                // Fallback Option - See [MotionEvent#Consistency Guarantees]
-                prevDownMotionEvent?.recycle()
-                prevDownMotionEvent = null
-            }
-        }.forceExhaustive
-
-        return null
+    fun onSelectKeyEvent(keyEvent: KeyEvent, pos: PointF) {
+        val event = MotionEvent.obtain(keyEvent.downTime, keyEvent.eventTime, keyEvent.action, pos.x, pos.y, 0)
```

Comment:
nit: I do kind of prefer `KeyEvent.toMotionEvent()` because it makes it declaratively clear that this is a pure conversion as opposed to having to read through each attribute.